mint-auditor/src/http_api/routes.rs

```diff
@@ -0,0 +1,43 @@
+use crate::http_api::{
+    api_types::{AuditorDb, BlockAuditDataResponse, CountersResponse},
+    service::MintAuditorHttpService,
+};
+use rocket::{get, serde::json::Json, State};
+
+/// index route
+#[get("/")]
+pub fn index() -> &'static str {
+    "Welcome to the mint auditor"
+}
+
+/// get counters
+#[get("/counters")]
+pub fn get_counters(db: &State<AuditorDb>) -> Result<Json<CountersResponse>, String> {
+    match MintAuditorHttpService::get_counters(&db.0) {
+        Ok(counters) => Ok(Json(CountersResponse::from(&counters))),
+        Err(e) => Err(e.to_string()),
+    }
+}
+
+/// Get the audit data for a target block
+#[get("/block_audit_data?<block_index>")]
```

Comment:
Consider making this a nested path, with `/` rather than `?`
```suggestion
#[get("/block_audit_data/<block_index>")]
```